### PR TITLE
[SPARK-21790][TESTS][FOLLOW-UP]  Add filter pushdown verification back.

### PR DESCRIPTION
**Pull request by wangyum**
## What changes were proposed in this pull request?

The previous PR(https://github.com/apache/spark/pull/19000) removed filter pushdown verification, This PR add them back.

## How was this patch tested?
manual tests

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala [hunk 1]
```diff
@@ -39,7 +39,6 @@ import org.apache.spark.sql.catalyst.plans.PlanTest
 import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 import org.apache.spark.sql.catalyst.util._
 import org.apache.spark.sql.execution.FilterExec
-import org.apache.spark.sql.internal.SQLConf
```
Comment (wangyum) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala]: Unused import

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 2]
```diff
@@ -255,6 +256,18 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
     val df = dfRead.filter(dfRead.col("date_type").lt(dt))
       .filter(dfRead.col("timestamp_type").lt(ts))
 
+    val parentPlan = df.queryExecution.executedPlan
+    assert(parentPlan.isInstanceOf[WholeStageCodegenExec])
+    val node = parentPlan.asInstanceOf[WholeStageCodegenExec]
+    val metadata = node.child.asInstanceOf[RowDataSourceScanExec].metadata
+    // The "PushedFilters" part should be exist in Datafrome's
```
Comment (gatorsmile) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: `Datafrome's ` -> `Dataframe's `

**Comment by gatorsmile** [comment]
LGTM except a minor comment.

**Comment by wangyum** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala (hunk 1 above)
Comment (HyukjinKwon) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala]: But I guess it's unrelated change. It wouldn't block this PR but I guess it's good to revert this one back.

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 3]
```diff
@@ -255,6 +256,18 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
     val df = dfRead.filter(dfRead.col("date_type").lt(dt))
       .filter(dfRead.col("timestamp_type").lt(ts))
 
+    val parentPlan = df.queryExecution.executedPlan
+    assert(parentPlan.isInstanceOf[WholeStageCodegenExec])
+    val node = parentPlan.asInstanceOf[WholeStageCodegenExec]
+    val metadata = node.child.asInstanceOf[RowDataSourceScanExec].metadata
+    // The "PushedFilters" part should be exist in Dataframe's
```
Comment (HyukjinKwon) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: little nit: `should be exist` -> `should exist`

File: external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala [hunk 4]
```diff
@@ -255,6 +256,18 @@ class OracleIntegrationSuite extends DockerJDBCIntegrationSuite with SharedSQLCo
...
+    // physical plan and the existence of right literals in
+    // "PushedFilters" is used to prove that the predicates
+    // pushing down have been effective.
+    assert(metadata.get("PushedFilters").ne(None))
```
Comment (HyukjinKwon) [inline external/docker-integration-tests/src/test/scala/org/apache/spark/sql/jdbc/OracleIntegrationSuite.scala]: nit: Could we use `isDefined` instead of `ne(None)`?

File: sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala (hunk 1 above)
Comment (wangyum) [inline sql/core/src/test/scala/org/apache/spark/sql/test/SQLTestUtils.scala]: OK, i'll revert it.

**Comment by wangyum** [comment]
retest this please

**Comment by gatorsmile** [comment]
Thanks! Merging to master.
